src/main/java/com/assignment/application/service/EmployeeServiceImpl.java

```diff
@@ -17,82 +18,59 @@
     @Autowired
     private EmployeeRepo employeeRepo;
 
+    @Autowired
+    private CachingInfo cachingInfo;
+
+    @Autowired
+    private CompanyRepo companyRepo;
+
+    @Autowired
+    private StringConstants stringConstants;
+
+
     @Override
-    public ResponseEntity<Employee> addEmployee(Long id, Employee employee) {
-        try{
-            if(employee.getCompanyId()!=id || employee.getId()==0){
-                return new ResponseEntity<>(null,HttpStatus.OK);
-            }
-            return new ResponseEntity<>(employeeRepo.save(employee),HttpStatus.OK);
-        }
-        catch (Exception e){
-            e.printStackTrace();
-            return new ResponseEntity<>(null, HttpStatus.INTERNAL_SERVER_ERROR);
+    public Employee addEmployee(Long companyId, Employee employee) {
+        Company company = companyRepo.findById(companyId).orElse(null);
+        if (employee == null || company == null || !employee.getCompanyId().equals(companyId) || employee.getId().equals(0)) {
+            return null;
         }
+        Employee employeeTemp = cachingInfo.addEmployee(employee,companyId);
```

Comment:
Save to repo first, and then cache the object, so that you have the right entity id in cached object.